src/NuGet.Core/NuGet.CommandLine.XPlat/Program.cs

```diff
@@ -247,7 +249,7 @@ public static int Main(string[] args)
             }
             catch (Exception e)
             {
-                EnsureLog(verbosity);
+                EnsureLog(GetLogLevel(verbosity));
```

Comment:
If any exception occurs outside of the restore command or before the restore command is executed, this `EnsureLog` is necessary. Note that this has to be run *after* `app.Execute` which takes the `string[] args` and parses them in to the various `CommandOption` instances.